src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useMemo, useState, useRef, useCallback, useEffect } from "react";
+import cx from "classnames";
+import PropTypes from "prop-types";
+import { SwitchTransition, CSSTransition } from "react-transition-group";
+
+import useEventListener from "../../../../hooks/useEventListener";
+import Icon from "../../../Icon/Icon";
+import Check from "../../../Icon/Icons/components/Check";
+
+import { MULTI_STEP_TYPES, STEP_STATUSES } from "../../MultiStepConstants";
+import { baseClassName } from "./StepIndicatorConstants";
+import "./StepIndicator.scss";
+import HiddenText from "../../../HiddenText/HiddenText";
+
+const StepIndicator = ({ stepComponentClassName, stepNumber, status, titleText, subtitleText, type }) => {
+  // Animations state
+  const [statusChangeAnimationState, setStatusChangeAnimationState] = useState(false);
+
+  // Refs
+  const componentRef = useRef(null);
+  const prevStatusRef = useRef(status);
+
+  // Callbacks for modifying animation state
+  const enableStatusChangeAnimation = useCallback(() => {
+    setStatusChangeAnimationState(true);
+  }, [setStatusChangeAnimationState]);
+
+  const disableStatusChangeAnimation = useCallback(() => {
+    setStatusChangeAnimationState(false);
+  }, [setStatusChangeAnimationState]);
+
+  const isStatusTransition = useCallback(() => prevStatusRef.current !== status, [prevStatusRef]);
+
+  // Event listeners for removing animation.
+  useEventListener({
+    eventName: "animationend",
+    callback: disableStatusChangeAnimation,
+    componentRef
+  });
+
+  // Effect - triggering animation when necessary.
+  useEffect(() => {
+    if (isStatusTransition()) {
+      enableStatusChangeAnimation();
+    }
+  }, [status]);
+
+  // Effect - updating previous status ref value (for animation) after component update.
+  useEffect(() => {
+    prevStatusRef.current = status;
+  }, [status]);
+
+  const ariaLabel = useMemo(() => {
+    return `Step ${stepNumber}: ${titleText} - ${subtitleText}, status: ${status}`;
+  }, [status]);
+
+  const baseClassNameWithType = `${baseClassName}--type-${type}`;
+  const baseClassNameWithStatus = `${baseClassName}--status-${status}`;
+  const baseClassNameWithAnimation = `${baseClassName}--with-animation`;
+
+  const getClassNamesWithSuffix = suffix => {
+    return [`${baseClassName}${suffix}`, `${baseClassNameWithType}${suffix}`, `${baseClassNameWithStatus}${suffix}`];
+  };
+
+  return (
+    <div
+      className={cx(...getClassNamesWithSuffix(""), stepComponentClassName, {
+        [baseClassNameWithAnimation]: statusChangeAnimationState
+      })}
+      aria-label={ariaLabel}
+    >
+      <div className={cx(...getClassNamesWithSuffix("__number-container"))} ref={componentRef}>
+        <SwitchTransition mode="out-in">
+          <CSSTransition
+            classNames={`${baseClassName}--swap`}
+            addEndListener={(node, done) => {
```

Comment:
cool, i didn't know it :)

Reply:
I pretty much copied this code from `Counter.jsx`, so I didn't know that too 😅